File: Prep-2025/prog/0028.cpp
#include<bits/stdc++.h>
using namespace std;
struct football{
    string name="";
    int score=0;
    int goal=0;
    int diff=0;
};
football team[4];
int arr[4][4];
bool compare(football a,football b){
    if(a.score!=b.score)return a.score>b.score;
    if(a.diff!=b.diff)return a.diff>b.diff;
    return a.goal>b.goal;
}
int main(){
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    for(int i=0;i<4;++i){
        cin>>team[i].name;
    }
    for(int i=0;i<4;++i){
        int sum=0;
        for(int j=0;j<4;++j){
            cin>>arr[i][j];
            sum+=arr[i][j];
        }
        team[i].goal=sum;
        team[i].diff=sum;
    }
    for(int j=0;j<4;++j){
        int sum=0;
        for(int i=0;i<4;++i){
            sum+=arr[i][j];
        }
        team[j].diff-=sum;
    }
    for(int i=0;i<4;++i){
        for(int j=i+1;j<4;++j){
            if(arr[i][j]>arr[j][i])team[i].score+=3;
            else if(arr[i][j]<arr[j][i])team[j].score+=3;
            else if(arr[i][j]==arr[j][i]){
                ++team[i].score;
                ++team[j].score;
            }
        }
    }
    sort(team,team+4,compare);
    for(int i=0;i<4;++i){
        cout<<team[i].name<<" "<<team[i].score<<"\n";
    }
    return 0;
}

File: python-prog/0028.py
import sys
class Football:
    def __init__(self,name="",score=0,goal=0,diff=0):
        self.name=name
        self.score=score
        self.goal=goal
        self.diff=diff
team=[Football() for i in range(4)]
arr=[[0]*4 for i in range(4)]
for i in range(4):
    team[i].name=sys.stdin.readline().strip()
for i in range(4):
    arr[i]=list(map(int,sys.stdin.readline().split()))
    s=sum(arr[i])
    team[i].goal=s
    team[i].diff=s
for j in range(4):
    s=0
    for i in range(4):
        s+=arr[i][j]
    team[j].diff-=s
for i in range(4):
    for j in range(i+1,4):
        if arr[i][j]>arr[j][i]:
            team[i].score+=3
        elif arr[i][j]<arr[j][i]:
            team[j].score+=3
        else:
            team[i].score+=1
            team[j].score+=1
team.sort(key=lambda x:(x.score,x.diff,x.goal),reverse=True)
for i in range(4):
    print(team[i].name,team[i].score)